seccompfilter.c

```diff
@@ -3,9 +3,30 @@
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
+#include <linux/seccomp.h>
 #include <seccomp.h>
 #include "seccomparch.h"
 
+#if defined(SCMP_ACT_KILL_PROCESS) && defined(SECCOMP_GET_ACTION_AVAIL) && \
+    defined(SECCOMP_RET_KILL_PROCESS)
+#include <unistd.h>
+#include <sys/syscall.h>
+
+uint32_t get_block_action()
+{
+    const uint32_t action = SECCOMP_RET_KILL_PROCESS;
+    /* Syscall fails if either actions_avail or kill_process is not available */
+    if (syscall(__NR_seccomp, SECCOMP_GET_ACTION_AVAIL, 0, &action) == 0)
+        return SCMP_ACT_KILL_PROCESS;
```

Comment:
NB: I haven't yet checked if a Ubuntu 20.04-based (i.e., one with libseccomp >=2.4) run still uses `SCMP_ACT_KILL_PROCESS` with this code.

Reply:
Just tried it out in a Ubuntu 20.04 VM. Works as intended with `SCMP_ACT_KILL_PROCESS`.